code/modules/clothing/chameleon/chameleon_gun.dm

```diff
@@ -54,36 +56,43 @@
 	inhand_x_dimension = gun_to_set.inhand_x_dimension
 	inhand_y_dimension = gun_to_set.inhand_y_dimension
 
+	// We dupe this casing and then delete it at the end, to grab the projectile.
+	var/obj/item/ammo_casing/casing_to_dupe
+
 	if(istype(gun_to_set, /obj/item/gun/ballistic))
 		var/obj/item/gun/ballistic/ball_gun = gun_to_set
+		// We also need to copy the starting magazine for ballistics.
 		var/obj/item/ammo_box/ball_ammo = new ball_gun.spawn_magazine_type(gun_to_set)
-		qdel(ball_gun)
 
 		if(!istype(ball_ammo) || !ball_ammo.ammo_type)
 			qdel(ball_ammo)
 			return FALSE
 
-		var/obj/item/ammo_casing/ball_cartridge = new ball_ammo.ammo_type(gun_to_set)
-		set_chameleon_ammo(ball_cartridge)
+		casing_to_dupe = new ball_ammo.ammo_type(gun_to_set)
+		set_chameleon_ammo(casing_to_dupe)
+		qdel(ball_ammo)
 
 	else if(istype(gun_to_set, /obj/item/gun/magic))
 		var/obj/item/gun/magic/magic_gun = gun_to_set
-		var/obj/item/ammo_casing/magic_cartridge = new magic_gun.ammo_type(gun_to_set)
-		set_chameleon_ammo(magic_cartridge)
+		casing_to_dupe = new magic_gun.ammo_type(gun_to_set)
```

Comment:
It's odd how half of these instantiate the casing in the contents of the gun, and the other half instantiate in the contents of src. should probably be consistent